src/test/java/dk/alexandra/fresco/lib/math/integer/stat/StatisticsTests.java

```diff
@@ -0,0 +1,201 @@
+							
+							SInt[][] covarianceMatrix = new SInt[3][3];
+							for (int i = 0; i < 3; i++) {
+								for (int j = 0; j < 3; j++) {
+									covarianceMatrix[i][j] = basicNumericFactory.getSInt();
+								}
+							}
+							
+							SInt[][] input = ioBuilder.inputMatrix(new int[][] {data1, data2, data3}, 1);
+							sequentialProtocolProducer.append(ioBuilder.getCircuit());
+							
+							MeanProtocol arithmeticMeanProtocol = statisticsFactory.getMeanProtocol(input[0], 10, mean1);
+							sequentialProtocolProducer.append(arithmeticMeanProtocol);
+
+							MeanProtocol arithmeticMeanProtocol2 = statisticsFactory.getMeanProtocol(input[1], 10, mean2);
+							sequentialProtocolProducer.append(arithmeticMeanProtocol2);
+
+							VarianceProtocol varianceProtocol = statisticsFactory.getVarianceProtocol(input[0], 10, mean1, variance);
+							sequentialProtocolProducer.append(varianceProtocol);
+							
+							CovarianceProtocol covarianceProtocol = statisticsFactory.getCovarianceProtocol(input[0], input[1], 10, mean1, mean2, covariance);
+							sequentialProtocolProducer.append(covarianceProtocol);
+							
+							CovarianceMatrixProtocol covarianceMatrixProtocol = statisticsFactory.getCovarianceMatrixProtocol(input, 10, covarianceMatrix);
+							sequentialProtocolProducer.append(covarianceMatrixProtocol);
+							
+							OInt output1 = ioBuilder.output(mean1);
+							OInt output2 = ioBuilder.output(mean2);
+							OInt output3 = ioBuilder.output(variance);
+							OInt output4 = ioBuilder.output(covariance);
+							OInt[][] output5 = ioBuilder.outputMatrix(covarianceMatrix);
+							
+							sequentialProtocolProducer.append(ioBuilder.getCircuit());
+							
+							ProtocolProducer gp = sequentialProtocolProducer;
+							
+							outputs = new OInt[] {output1, output2, output3, output4, output5[0][0], output5[1][0]};
+							
+							return gp;
+						}
+					};
+					sce.runApplication(app);
+					BigInteger mean1 = app.getOutputs()[0].getValue();
+					BigInteger mean2 = app.getOutputs()[1].getValue();
+					BigInteger variance = app.getOutputs()[2].getValue();
+					BigInteger covariance = app.getOutputs()[3].getValue();
+					
+					double sum = 0.0;
+					for (int entry : data1) {
+						sum += entry;
+					}
+					double mean1Exact = sum / data1.length;
+							
+					sum = 0.0;
+					for (int entry : data2) {
+						sum += entry;
+					}
+					double mean2Exact = sum / data2.length;
+					
+					
+					double ssd = 0.0;
+					for (int entry : data1) {
+						ssd += (entry - mean1Exact) * (entry - mean1Exact);
+					}
+					double varianceExact = ssd / (data1.length - 1);
+					
+					double covarianceExact = 0.0;
+					for (int i = 0; i < data1.length; i++) {
+						covarianceExact += (data1[i] - mean1Exact) * (data2[i] - mean2Exact);
+					}
+					covarianceExact /= (data1.length - 1);
+					
+					System.out.println("Calculated mean1 = " + mean1 + " (exact value is " + mean1Exact + ")");
```

Comment:
Please do not have tests print out a lot of messages. Also make sure to make test assertions that your test does what it is supposed to.